### PR TITLE
orchestrator-process: store PID metadata in temp directory

### PR DESCRIPTION
**Pull request by benesch**
PID files are not valid after a reboot of a machine. In the best case, the referenced PIDs do not exist, and the process orchestrator correctly recreates the services; in the worst case, the PIDs have been reused by different processes entirely, and the process orchestrator incorrectly thinks the services are already running. The worst case scenario is almost a guarantee with containers, where there are only a few processes using the low-numbered containers.

This commit fixes the problem by moving  the PID metadata files into $TMPDIR/environment-$ID. $TMPDIR is cleared on restart, so the stale PID files will correctly vanish after a restart. Naming the directory after the environment ID ensures that environmentd can find its metadata after a process restart without a machine restart, but allows multiple `environmentd` processes to co-exist, as long as they use different environment IDs. Things work correctly with the `--reset` option to bin/environment, too, as this option generates a new environmentd ID.

Touches #15725.
Would close #15800.

### Motivation

  * This PR takes a step towards fixing a recognized bug.

### Checklist

- [x] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way) and therefore is tagged with a `T-proto` label.
- [x] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):

  - n/a

### CODE REVIEW

**Review by guswynn — APPROVED** [review]
ci looks mad but this was shockingly painless to do it appears!

**Comment by benesch** [comment]
Yep, _this_ part was easy. Sadly the rest of https://github.com/MaterializeInc/materialize/issues/15725 is hard!

File: src/environmentd/src/bin/environmentd/main.rs [hunk 1]
```diff
@@ -295,10 +295,14 @@ pub struct Args {
...
-    /// Where the process orchestrator should store its metadata.
+    /// Where the process orchestrator should store secrets.
+    // TODO(benesch): the name "data directory" is historical. We should rename
+    // to "--orchestrator-process-secrets-directory", but there are some
+    // irritating backwards compatibility concerns to sort out with the upgrade
+    // test.
     #[clap(
         long,
-        env = "ORCHESTRATOR_PROCESSDATA_DIRECTORY",
+        env = "ORCHESTRATOR_PROCESS_DATA_DIRECTORY",
```
Comment (philip-stoev) [inline src/environmentd/src/bin/environmentd/main.rs:305]: This variable was used with its old name in mzcompose/services.py . I guess, to be safe, services.py needs to set the variable with both its old and new spelling.

**Comment by philip-stoev** [comment]
I pushed https://github.com/MaterializeInc/materialize/pull/15800 as a temporary work-around until this is sorted out.

**Comment by benesch** [comment]
@jkosh44 just a heads up that the process orchestrator looks _very_ different as a result of this PR. So that you're not surprised when you next look at it.

**Pull request merged**